flocker/control/_model.py

```diff
@@ -376,7 +379,19 @@ def __invariant__(self):
                     return (False, '%r manifestation is not on node' % (app,))
         return (True, "")
 
+    def __new__(cls, **kwargs):
+        # PRecord does some crazy stuff, thus _precord_buckets; see
+        # PRecord.__new__.
+        if "uuid" not in kwargs and "_precord_buckets" not in kwargs:
+            warn("UUID is required, this is for backwards compat with existing"
+                 " tests only. If you see this in production code that's "
+                 "a bug.", DeprecationWarning, stacklevel=2)
+            kwargs["uuid"] = ip_to_uuid(kwargs["hostname"])
+        return PRecord.__new__(cls, **kwargs)
```

Comment:
:question: 
Possibly could do this by handling an `InvariantException` from `PRecord.__new__` and then re-trying with the updated keyword arguments.  Possibly better if it isolates us from implementation details like `_precord_buckets` (and avoids a change in pyrsistent breaking our assumptions here).  On the other hand, pinned versions!  **Hooray**.